packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -124,7 +124,9 @@ class ServiceConnectionManager {
     }
 
     this.service = service;
-    serviceAvailable.complete();
+    if (!serviceAvailable.isCompleted) {
+      serviceAvailable.complete();
```

Comment:
why are we completing this object multiple times? perhaps add a TODO to track down the underlying bug?

Reply:
thanks for the writeup. That sounds like we are accidentally creating a coulple of ServiceConnectionManager objects. Can we fix the code to avoid that? For example, add a serviceLoading completer so that we don't get multiple simultaneous loads.
For example: currently it sounds like we are listening for all these events twice when we should only be doing it once.

Reply:
Done.